core/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -515,11 +513,17 @@ private Snapshot createSnapshot(long minGeneration) {
      * while receiving future ones as well
      */
     public Translog.View newView() {
+        long viewGenToClean = -1;
         try (ReleasableLock lock = readLock.acquire()) {
             ensureOpen();
-            View view = new View(lastCommittedTranslogFileGeneration);
-            outstandingViews.add(view);
+            viewGenToClean = deletionPolicy.acquireTranslogGenForView();
```

Comment:
I really do not like the use of setting `viewGenToClean` to `-1` to indicate not to release the view. Is there a reason that you do not make `viewGenToClean` final and local to the try block and set a boolean flag to indicate success or not?